apps/beeswax/src/beeswax/server/dbms.py

```diff
@@ -462,13 +462,24 @@ def _get_tables_via_sparksql(self, database, table_names='*'):
       self.close(handle)
 
       # We get back: database | tableName | isTemporary
-      return [{
-          'name': row[1],
-          'type': 'VIEW' if row[2] else 'TABLE',
-          'comment': ''
-        }
-        for row in result.rows()
-      ]
+      try:
+        return [{
+            'name': row[1],
+            'type': 'VIEW' if row[2] else 'TABLE',
+            'comment': ''
+          }
+          for row in result.rows()
+        ]
+      except:
+        tables = result.data_table.results.results.columns[0].stringVal.values
+        # We get back: database | tableName | isTemporary
+        return [{
+            'name': table,
+            'type': 'TABLE',
```

Comment:
To check for `VIEWS` also? Just like in line 468?

Reply:
while checking the code wat is the conceptual difference between views and tables?

Reply:
I think this post covers it well: https://stackoverflow.com/questions/6015175/difference-between-view-and-table-in-sql